### PR TITLE
Add Australian Company Number validator

### PR DESCRIPTION
**Pull request by koterpillar**
Add [Australian Company Number](https://en.wikipedia.org/wiki/Australian_Company_Number) model and form fields.

Note that running `isort --recursive --line-width 120 localflavor tests` produced changes in otherwise unchanged files. I've limited my changes to the files I touched.

**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

- [x] Adjust your imports to a standard form by running this command:

      `isort --recursive --line-width 120 localflavor tests`

**New Fields Only**

- [x] Prefix the country code to all fields.

- [x] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [x] Prefer '<country code>PostalCodeField' for postal codes as it's
      international English; ZipCode is a term specific to the United
      States postal system.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [x] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [x] Add documentation for all fields.

### CODE REVIEW

File: tests/test_au/tests.py [hunk 1]
```diff
@@ -195,6 +196,56 @@ def test_raises_error_for_invalid_abn(self):
...
+
+    def test_raises_error_for_too_long_acn(self):
+        """Test an ACN with more than nine digits is invalid."""
+        invalid_acn = '6043275040'
+        validator = AUCompanyNumberFieldValidator()
+        self.assertRaises(ValidationError, lambda: validator(invalid_acn))
+
+    def test_raises_error_for_whitespace(self):
+        """Test an ACN can be valid when it contains whitespace."""
+        # NB: Form field should strip the whitespace before regex valdation is run.
```
Comment (benkonrath) [inline tests/test_au/tests.py]: valdation -> validation

I see that's also in the copy & pasted code so maybe you can also fix the typo there as well.

As this comment points out, the whitespace is stripped in the form. It would be nice to have a test for the form as well - at least testing the expected functionality that isn't tested with the validation tests.

File: tests/test_au/tests.py [hunk 2]
```diff
@@ -195,6 +196,56 @@ def test_raises_error_for_invalid_abn(self):
...
 
+class AULocalFlavorAUCompanyNumberFieldValidatorTests(TestCase):
+
+    def test_no_error_for_a_valid_acn(self):
+        """Test a valid ACN does not cause an error."""
+        valid_acn = '604327504'
+
+        validator = AUCompanyNumberFieldValidator()
+
+        validator(valid_acn)
```
Comment (benkonrath) [inline tests/test_au/tests.py]: It would be nice to be consistent on the spacing. For instance, some tests have a space between each statement while one has no spaces. The space don't seem to be needed because it's pretty simple code. Feel free to fix the previous tests as well where this was copied from.

**Comment by koterpillar** [comment]
Fixed both issues.

**Comment by benkonrath** [comment]
It would be great if you can also add a test the form's `prepare_value`. I think there's an example test with the `AUBusinessNumberField`. Thanks.

**Comment by koterpillar** [comment]
Added tests.
